src/control/lib/hardware/sysfs/provider.go

```diff
@@ -191,10 +190,54 @@ func (s *Provider) getPCIAddress(path string) (*hardware.PCIAddress, error) {
 		return nil, errors.Wrap(err, "couldn't get PCI device")
 	}
 
-	pciAddr, err := hardware.NewPCIAddress(filepath.Base(pciPath))
+	var pciAddr *hardware.PCIAddress
+	for pciPath != s.sysPath("devices") {
+		pciAddr, err = hardware.NewPCIAddress(filepath.Base(pciPath))
+		if err == nil {
+			return pciAddr, nil
+		}
+
+		pciPath = filepath.Dir(pciPath)
+	}
+
+	return nil, errors.Errorf("unable to parse PCI address from %q", path)
+}
+
+// GetFabricInterfaces harvests the CXI fabric interfaces from sysfs.
```

Comment:
It's true that for the moment, we're only harvesting CXI interfaces using sysfs, but I'm not sure that we should make the API documentation specific to that use case.

Reply:
Good point. I can adjust this in my next PR, which is in the works at the moment.